src/store/cache/proposals/hooks.ts

```diff
@@ -195,15 +212,90 @@ export const useVotingPeriod = () => {
   return { votingPeriod }
 }
 
+export const useExecutionDelay = () => {
+  const executionDelay = useSelector(getExecutionDelay)
+  const dispatch = useDispatch()
+  useEffect(() => {
+    if (!executionDelay) {
+      dispatch(fetchExecutionDelay())
+    }
+  }, [dispatch, executionDelay])
+
+  return { executionDelay }
+}
+
 export const useProposalTimeRemaining = (submissionBlock: number) => {
   const { votingPeriod } = useVotingPeriod()
   const remaining = useTimeRemaining(submissionBlock, votingPeriod)
   return remaining
 }
 
+export const useExecutionDelayTimeRemaining = (votingDeadlineBlock: number) => {
```

Comment:
should we use these hooks in the <Proposal> list view component?
https://github.com/AudiusProject/protocol-dashboard/blob/main/src/components/Proposal/Proposal.tsx